editoast/src/modelsv2/stdcm_search_environment.rs

```diff
@@ -0,0 +1,199 @@
+use chrono::NaiveDateTime;
+use diesel::ExpressionMethods;
+use diesel::QueryDsl;
+use diesel_async::RunQueryDsl;
+use serde::Serialize;
+use utoipa::ToSchema;
+
+use crate::error::Result;
+use crate::modelsv2::prelude::{Create, Row};
+use editoast_derive::ModelV2;
+use editoast_models::DbConnection;
+
+#[cfg(test)]
+use serde::Deserialize;
+
+#[derive(Debug, Clone, ModelV2, ToSchema, Serialize)]
+#[model(table = crate::tables::stdcm_search_environment)]
+#[cfg_attr(test, derive(Deserialize, PartialEq), model(changeset(derive(Clone))))]
+pub struct StdcmSearchEnvironment {
+    pub id: i64,
+    pub infra_id: i64,
+    #[schema(nullable = false)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub electrical_profile_set_id: Option<i64>,
+    #[schema(nullable = false)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub work_schedule_group_id: Option<i64>,
+    pub timetable_id: i64,
+    pub search_window_begin: NaiveDateTime,
+    pub search_window_end: NaiveDateTime,
+}
+
+impl StdcmSearchEnvironment {
+    /// Retrieve the latest search environment. Returns None if no search environment is found.
+    pub async fn retrieve_latest(conn: &mut DbConnection) -> Option<Self> {
+        use crate::tables::stdcm_search_environment::dsl::*;
+        stdcm_search_environment
+            .order_by((search_window_end.desc(), search_window_begin.asc()))
+            .first::<Row<StdcmSearchEnvironment>>(conn)
+            .await
+            .map(Into::into)
+            .ok()
+    }
+
+    pub async fn delete_all(conn: &mut DbConnection) -> Result<()> {
+        use crate::tables::stdcm_search_environment::dsl::*;
+        diesel::delete(stdcm_search_environment)
+            .execute(conn)
+            .await?;
+        Ok(())
+    }
+}
+
+impl StdcmSearchEnvironmentChangeset {
+    pub async fn over_write(self, conn: &mut DbConnection) -> Result<StdcmSearchEnvironment> {
```

Comment:
I believe overwrite is one word.

```suggestion
    pub async fn overwrite(self, conn: &mut DbConnection) -> Result<StdcmSearchEnvironment> {
```